### ISSUE TITLE
Add option to disable the extra clang flags in MacOS

### ISSUE THREAD

**Issue by Azure-Vani** [ycm-core/ycmd#1340]
In particular, the `if add_extra_clang_flags:` branch in `PrepareFlagsForClang` function in file ycmd/ycmd/completers/cpp/flags.py. Otherwise, the result when complete headers is super noisy when somebody uses g++ rather than clang on macOS

**Comment by bstaletic** [comment]
I don't think that is a good idea. That branch that reaches into `AddMacIncludePaths` is also responsible for adding `-resource-dir=` and `-fno-delayed-template-parsing` on Windows. I've done invasive changes in `flags.py` before and it was very frustrating.

That said, you can still avoid that by:

- Switching to the clangd based completer
- Using compilation databases instead of the extra conf files.

**Comment by puremourning** [comment]
I actually had reason to need this recently myself for some very bizarre reasons so maybe I will add it at some point.
